### PR TITLE
Benchmarking

### PR DESCRIPTION
**Pull request by juankysoriano**
The following PR includes a `benchmark` module comparing three different XML parsers:

* [SEXP](https://github.com/novoda/simple-easy-xml-parser)
* [Jackson](https://github.com/FasterXML/jackson-dataformat-xml)
* [Simple-Framework](http://simple.sourceforge.net/)

The benchmark is performed for two different sets of data, `small.xml` and `medium.xml`.

screen shot 2015-12-27 at 17 55 18

The results of the benchmarking gets special relevance by analysing the `medium` data set. Our benchmarking shows that *SEXP is performing faster than it's competitors*:
 
**1.7x faster than [Jackson](https://github.com/FasterXML/jackson-dataformat-xml)**
**2.5x faster than [Simple-Framework](http://simple.sourceforge.net/)**, 

Further *SEXP is way more efficient* in terms of memory allocation, which could be a more than reasonable argument to make use of it in your mobile applications. 

**NOTE**: The benchmarking has been performing using [Caliper](https://github.com/google/caliper); Google’s open-source framework for writing, running and viewing the results of Java Microbenchmarks. We understand our tests have been ran on a JVM and this could influence the results vs Dalvik. In the future we will be also running an Android benchmark.

### CODE REVIEW

**Comment by juankysoriano** [comment]
@blundell santa claus is coming!

**Comment by blundell** [comment]
:100:

**Comment by ouchadam** [comment]
awesome stuff :+1:  

simplexml makes a whole lot of objects. Must be autoboxing?

**Comment by xrigau** [comment]
maybe @ouchadam and it also has to create all the objects for reflection so that will add too!

**Comment by amlcurran** [comment]
How about a large file as well? The files you guys use are generally a fair bit larger than the two examples benchmarked here

File: benchmark/src/main/resources/medium.xml [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    <title type="text">Sport</title>
+    <summary type="html">Sport Programmes</summary>
+    <updated>2015-03-13T10:49:48.144Z</updated>
+    <link href="http://google.com" rel="self" title="A" type="application/atom+xml" />
+    <link href="http://google.com" rel="related" title="B" type="application/atom+xml" />
+    <link href="http://google.com" rel="related" title="C" type="application/atom+xml" />
+    <link href="http://google.com" rel="related" title="D" type="application/atom+xml" />
+    <link href="http://google.com" rel="related" title="E" type="application/atom+xml" />
+    <link href="http://google.com" rel="related" title="F" type="application/atom+xml" />
+  </entry>
```
Comment (blundell) [inline benchmark/src/main/resources/medium.xml]: i thought we'd copy pasted this entry element a few more times?

Comment (juankysoriano) [inline benchmark/src/main/resources/medium.xml]: Pushing 100 entries to `medium.xml`

File: README.md [hunk 2]
```diff
@@ -5,7 +5,9 @@ XML parsing is now sexy!
 
 ## Description
 
-A simple typed XML parser based upon Android sax parser but written in pure Java. SEXP gives callbacks for all parsing events and being written in pure java allows faster and more comprehensive testability.
+A simple, high performance, typed XML parser based upon Android sax parser but written in pure Java. SEXP gives callbacks for all parsing events and being written in pure java allows faster and more comprehensive testability.
+
+Moreover SEXP is a pull-parsing XML deserialiser which affects both performance (no reflection is required) and low memory footprint (unncessary objects are not allocated). Our testing shows that SEXP is performing 1.7x faster than [Jackson](https://github.com/FasterXML/jackson-dataformat-xml) and 2.5x faster than [Simple-Framework](http://simple.sourceforge.net/). You can refer to the following [micro-benchmarking](https://github.com/novoda/simple-easy-xml-parser/tree/master/benchmark) for further information.
```
Comment (juankysoriano) [inline README.md]: Added some number to the performance gaining here.

**Comment by juankysoriano** [comment]
@blundell @amlcurran I have modified the `medium.xml` set to parse 100 entry elements. 

With this test SEXP is revealing itself as a very fast and efficient XML parser as it parses 2.5x faster than Simple-Framework and 1.7x faster than Jackson.

**Comment by ouchadam** [comment]
where are these time taken numbers?

**Comment by juankysoriano** [comment]
@ouchadam There is a screenshot in the PR description showing numbers. The screenshot is taken from here:

https://microbenchmarks.appspot.com/runs/39f714bc-544b-4c87-8b2e-fafcfae68e84#r:scenario.vmSpec.options.BackgroundCompilation,scenario.vmSpec.options.UseCompiler,scenario.vmSpec.options.UseFastAccessorMethods,scenario.vmSpec.options.UseFastEmptyMethods,scenario.vmSpec.options.UseLoopCounter,scenario.vmSpec.options.UseOnStackReplacement,scenario.benchmarkSpec.methodName,scenario.vmSpec.properties.java.vm.info

**Comment by ouchadam** [comment]
I thought runtime was still objects! oops

**Comment by blundell** [comment]
:+1: LGTM

**Comment by xrigau** [comment]
yayy :+1: good news!

**Pull request merged**